tests/test_rich_markup_mode.py

```diff
@@ -38,3 +41,186 @@ def main(arg: str):
+                "",
+            ],
+        ),
+        (
+            "rich",
+            [
+                "This header is just pretty long really",
+                "Line 1 of a very",
+                "extremely super long",
+                "line",
+                "",
+                "And a short line 2",
+                "",
+            ],
+        ),
+        (
+            "none",
+            [
+                "This header is just pretty long really",
+                "Line 1 of a very extremely super long line",
+                "And a short line 2",
+                "",
+            ],
+        ),
+    ],
+)
+def test_markup_mode_newline_issue447(mode: str, lines: List[str]):
+    app = typer.Typer(rich_markup_mode=mode)
+
+    @app.command()
+    def main(arg: str):
+        """This header
+        is just
+        pretty long
+        really
+
+        Line 1 of a very
+        extremely super long
+        line
+
+        And a short line 2
+        """
+        print(f"Hello {arg}")
+
+    assert app.rich_markup_mode == mode
+
+    result = runner.invoke(app, ["World"])
+    assert "Hello World" in result.stdout
+
+    result = runner.invoke(app, ["--help"])
+    result_lines = [line.strip() for line in result.stdout.split("\n")]
+    assert any(c in result.stdout for c in rounded)
+    help_start = [i for i, row in enumerate(result_lines) if "This header" in row][0]
+    arg_start = [i for i, row in enumerate(result_lines) if "Arguments" in row][0]
+    assert help_start != -1
+    assert result_lines[help_start:arg_start] == lines
+
+
+@pytest.mark.parametrize(
+    "mode,lines",
+    [
+        pytest.param(
+            "markdown",
+            ["First line", "", "• 1", "• 2", "• 3", ""],
+            marks=pytest.mark.xfail(),
+        ),
+        ("rich", ["First line", "- 1", "- 2", "- 3", ""]),
+        ("none", ["First line", "- 1 - 2 - 3", ""]),
+    ],
+)
+def test_markup_mode_bullets(mode: str, lines: List[str]):
+    app = typer.Typer(rich_markup_mode=mode)
+
+    @app.command()
+    def main(arg: str):
+        """First line
+
+        - 1
+        - 2
+        - 3
+        """
+        print(f"Hello {arg}")
+
+    assert app.rich_markup_mode == mode
+
+    result = runner.invoke(app, ["World"])
+    assert "Hello World" in result.stdout
+
+    result = runner.invoke(app, ["--help"])
+    result_lines = [line.strip() for line in result.stdout.split("\n")]
+    assert any(c in result.stdout for c in rounded)
+    help_start = result_lines.index("First line")
+    arg_start = [i for i, row in enumerate(result_lines) if "Arguments" in row][0]
+    assert help_start != -1
+    assert result_lines[help_start:arg_start] == lines
+
+
+@pytest.mark.parametrize(
+    "mode,lines",
+    [
+        pytest.param(
+            "markdown",
+            ["First line", "", "• 1", "• 2", "• a", "• b", "• 3", ""],
+            marks=pytest.mark.xfail(),
+        ),
```

Comment:
This test currently fails on `master`, with the output being 
```
['First line', '', '• 1 - 2   - a   - b - 3', '']
```
-> this happens mostly because the newlines are not parsed correctly, after which the bullet points aren't recognized correctly. PR #815 attempts to fix this.